File: examples/go/mail/mail.go
package main

import (
	"encoding/json"
	"net/smtp"
	"os"
	"strings"

	"github.com/thcdrt/funk-go/event"
)

type Data struct {
	To      []string `json:"to"`
	Subject string   `json:"subject"`
	Text    string   `json:"text"`
}

func Send(event event.Event) (string, error) {
	host := os.Getenv("MAIL_SMTP_HOST")
	port := os.Getenv("MAIL_SMTP_PORT")
	username := os.Getenv("MAIL_SMTP_USERNAME")
	password := os.Getenv("MAIL_SMTP_PASSWORD")

	var data Data
	err := json.Unmarshal([]byte(event.Data), &data)
	if err != nil {
		return "", err
	}

	msg := []byte("To: " + strings.Join(data.To[:], " ") + "\r\n" +
		"Subject: " + data.Subject + "\r\n" +
		"\r\n" +
		data.Text + "\r\n",
	)

	auth := smtp.PlainAuth("", username, password, host)
	err = smtp.SendMail(host+":"+port, auth, username, data.To, msg)
	if err != nil {
		return "", err
	}

	return "Mail sent", nil
}

File: examples/perl/mail/mail.pl
use Mail::Sendmail;

	$event->{'data'}{'to'} = "couderthomas\@gmail.com";
	$event->{'data'}{'subject'} = "coucou";
	$event->{'data'}{'text'} = "rololo";
	my $from = "yo\@functions.ovh";

	print $event->{'data'}{'to'};

	%mail = ( 
		'Smtp'    => "yo%40functions.ovh:houbahouba\@ssl0.ovh.net",
		'To'      => $event->{'data'}{'to'},
        'From'    => $from,
        'Subject' => $event->{'data'}{'subject'},
        'Message' => $event->{'data'}{'text'}
    );

	sendmail(%mail) or die $Mail::Sendmail::error;
	print $Mail::Sendmail::log."\n";